### ISSUE TITLE
How can we use fastlane from another gem

### ISSUE THREAD

**Issue by elitalon** [fastlane/fastlane#11743]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Let's say I'm writing a gem that requires fastlane and offers a DSL on top of its built-in actions. If I wanted to invoke for instance [register_devices](https://github.com/fastlane/fastlane/blob/765eb2ac93e5e2fbb61c701b4a75a3135c812a7b/fastlane/lib/fastlane/actions/register_devices.rb) I could do this:

```ruby
require 'fastlane'
Fastlane.load_actions

def register_team_devices(devices:)
  params = {
    devices: devices,
    # set other parameters here…
  }
  Fastlane::Actions::RegisterDevicesAction.run(params)
end
```

But I was wondering if there is a more straightforward way to access the whole functionality of fastlane, as if we were in the context of a Fastfile. I browsed the source code, but the project is big and I'm still a bit rusty in Ruby.

Thanks for your help.

### Environment
Not relevant for the question.

**Comment by KrauseFx** [comment]
Hey @elitalon, that's a great question. Yep your code looks correct 👍  There is no real easier way. You could also generate a `Fastfile` on the fly and use that with the `fastlane/runner` I guess? 

All of that is internal API though, so it might change in the future (probably not soon, but you know, maybe)

**Comment by elitalon** [comment]
I think I will stick to the public API. It's not that much code and I think it would be less arcane for other people in our team to understand.

Thanks @KrauseFx! I'm closing this 😉

**Comment by KrauseFx** [comment]
Awesome 👍 What are you building?

**Comment by elitalon** [comment]
Well, our team works in many apps from different customers (we are a small agency). We need to manage a lot of devices, provision different flavours of every app, trigger builds in our CI,… The usual stuff.

We've been able to manage everything with a small Fastfile per project, plus a big shared Fastfile. But things have started to get a bit out of hand, specially in the shared Fastfile. The complexity of the lanes went up and we have duplicated logic that we want to eliminate.

We tried moving code from the lanes into helper functions and files. But organising the code is not so easy: `import` and `require` work locally, while `import_from_git` requires several checkouts to load everything (as far as I know).

We are considering now replacing the shared Fastfile with an internal gem that we can use from the per-project Fastfiles. I'm not entirely sure if that extra layer of abstraction will help, right now we're only experimenting.

I would appreciate if you have any suggestion on this 😅
